my-sms/app/controllers/institutes_controller.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: false
+
+class InstitutesController < ApplicationController
+  decorates_assigned :institute
+
+  before_filter :load_institute, except: %i[index]
+  before_render :flash_errors, only: %i[edit create update]
```

Comment:
Do you need flash_errors on `create` ?
What happens when you force an error on a new Institute ?

Reply:
Its a bit hard to test on institute as the actual edit/create/update functionality isn't included until the next task. I mainly included the `flash_errors` here to scaffold out the controller in preparation for that.

I did test it on student, which has the same functionality, and it works. `Create` redirects back to the new page if there's any errors on the model.
![image](https://user-images.githubusercontent.com/41467557/108917097-57288a00-7683-11eb-9355-cfaf091abfb7.png)

Reply:
Interesting, so it fails and redirects back to the new method in the controller.
However the new method is not in `before_render :flash_errors, only:  %i[edit create update]`
How then is the flash_error showing for the new page ?

Reply:
Its not explicitly a redirect, the controller just renders the `'new'` action. The before_render acts as soon as `render` is called so it doesn't matter which action gets rendered, just which route is called by the dispatcher.